src/widgets/block.rs

```diff
@@ -396,16 +400,38 @@ impl<'a> Block<'a> {
         self
     }
 
-    /// Defines the block style.
+    /// Defines the style of the entire block.
     ///
     /// This is the most generic [`Style`] a block can receive, it will be merged with any other
-    /// more specific style. Elements can be styled further with [`Block::title_style`] and
-    /// [`Block::border_style`].
+    /// more specific styles. Elements can be styled further with [`Block::title_style`] and
+    /// [`Block::border_style`], which will be applied on top of this style. If the block is used
+    /// as a container for another widget, the inner widget can also be styled.
     ///
     /// `style` accepts any type that is convertible to [`Style`] (e.g. [`Style`], [`Color`], or
     /// your own type that implements [`Into<Style>`]).
     ///
-    /// This will also apply to the widget inside that block, unless the inner widget is styled.
+    /// This will also apply to the widget inside that block, the inner widget overrides the style.
+    ///
+    /// See [`Style`] for more information on how merging styles works.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use ratatui::{prelude::*, widgets::*};
+    /// let block = Block::new().style(Style::new().red().on_black());
+    ///
+    /// // To override styles set on the block level, you can style the title, border.
```

Comment:
The example is way more helpful than the current sentence here. I still think override is a confusing word. unset styles or add partially to override seems better to me, as that doesnt imply the style "below" is ignored then.
Alternatively, remove the sentence in front of this code block completely as the description above explains them to be applied on top of the block style already.

```suggestion
    /// // For border and title you can partially override / unset styles set on the block level.
```